superset/viz.py

```diff
@@ -947,9 +947,9 @@ def query_obj(self):
         return d
 
     def get_data(self, df):
-        df['x'] = df[[self.x_metric]]
-        df['y'] = df[[self.y_metric]]
-        df['size'] = df[[self.z_metric]]
+        df['x'] = df[[utils.get_metric_name(self.x_metric)]]
```

Comment:
Any reason not to set self.x_metric to utils.get_metric_name(form_data.get('x'))? Is the full metric needed anywhere?

Reply:
Good question. The metric needs to be fully defined for the SQLA progressing as there's a validation check to see whether or not it's an [ad-hoc metric](https://github.com/apache/incubator-superset/blob/master/superset/connectors/sqla/models.py#L534), hence it's only at data extraction that we can reference the metric by name.